--- ags/app/machine/ags_drum_callbacks.c.orig	2022-02-22 17:35:10.498071966 +0100
+++ ags/app/machine/ags_drum_callbacks.c	2022-02-22 17:38:24.272330626 +0100
@@ -90,9 +90,13 @@
   GtkFileChooserDialog *file_chooser;
   GtkCheckButton *check_button;
 
+  gchar *uri;
+  
   if(drum->open_dialog != NULL){
     return;
   }
+
+  uri = g_strdup_printf("file://%s/usr/share/hydrogen/data/drumkits", getenv("AGS_SNAP_PATH"));
   
   file_chooser = (GtkFileChooserDialog *) gtk_file_chooser_dialog_new(g_strdup("open audio files"),
 								      (GtkWindow *) gtk_widget_get_toplevel((GtkWidget *) drum),
@@ -101,7 +105,7 @@
 								      i18n("_Cancel"), GTK_RESPONSE_CANCEL, 
 								      NULL);
   gtk_file_chooser_add_shortcut_folder_uri(GTK_FILE_CHOOSER(file_chooser),
-					   "file:///usr/share/hydrogen/data/drumkits",
+					   uri,
 					   NULL);  
   drum->open_dialog = (GtkWidget *) file_chooser;
   gtk_file_chooser_set_select_multiple(GTK_FILE_CHOOSER(file_chooser), TRUE);
@@ -122,6 +126,8 @@
 		   G_CALLBACK(ags_drum_open_response_callback), drum);
   g_signal_connect(G_OBJECT(file_chooser), "response",
 		   G_CALLBACK(ags_machine_open_response_callback), drum);
+
+  g_free(uri);
 }
 
 void
--- ags/app/machine/ags_drum_input_pad_callbacks.c.orig	2022-02-22 17:38:51.440576904 +0100
+++ ags/app/machine/ags_drum_input_pad_callbacks.c	2022-02-22 17:39:31.156890675 +0100
@@ -43,9 +43,14 @@
   GtkSpinButton *spin_button;
 //  GtkToggleButton *play;
 
+  gchar uri;
+  
   if(drum_input_pad->file_chooser != NULL)
     return;
 
+  uri = g_strdup_printf("file://%s/usr/share/hydrogen/data/drumkits", getenv("AGS_SNAP_PATH"));
+  
+  
   drum_input_pad->file_chooser =
     file_chooser = (GtkFileChooserDialog *) gtk_file_chooser_dialog_new(i18n("Open File"),
 									(GtkWindow *) gtk_widget_get_toplevel((GtkWidget *) drum_input_pad),
@@ -54,7 +59,7 @@
 									i18n("_Open"), GTK_RESPONSE_ACCEPT,
 									NULL);
   gtk_file_chooser_add_shortcut_folder_uri(GTK_FILE_CHOOSER(file_chooser),
-					   "file:///usr/share/hydrogen/data/drumkits",
+					   uri,
 					   NULL);
   gtk_file_chooser_set_select_multiple(GTK_FILE_CHOOSER(file_chooser),
 				       FALSE);
@@ -98,6 +103,8 @@
 
   g_signal_connect((GObject *) file_chooser, "response",
 		   G_CALLBACK(ags_drum_input_pad_open_response_callback), (gpointer) drum_input_pad);
+
+  g_free(uri);
 }
 
 void
--- ags/app/machine/ags_ffplayer_callbacks.c.orig	2022-02-22 17:35:03.485878621 +0100
+++ ags/app/machine/ags_ffplayer_callbacks.c	2022-02-22 17:37:37.335794276 +0100
@@ -237,9 +237,13 @@
 {
   GtkFileChooserDialog *file_chooser;
 
+  gchar *uri;
+
+  uri = g_strdup_printf("file://%s/usr/share/sounds/sf2", getenv("AGS_SNAP_PATH"));
+  
   file_chooser = ags_machine_file_chooser_dialog_new(AGS_MACHINE(ffplayer));
   gtk_file_chooser_add_shortcut_folder_uri(GTK_FILE_CHOOSER(file_chooser),
-					   "file:///usr/share/sounds/sf2",
+					   uri,
 					   NULL);
   ffplayer->open_dialog = (GtkWidget *) file_chooser;
   gtk_file_chooser_set_select_multiple(GTK_FILE_CHOOSER(file_chooser),
@@ -249,6 +253,8 @@
 		   G_CALLBACK(ags_ffplayer_open_dialog_response_callback), AGS_MACHINE(ffplayer));
 
   gtk_widget_show_all((GtkWidget *) file_chooser);
+
+  g_free(uri);
 }
 
 void
